pydantic_evals/pydantic_evals/evaluators/_run_evaluator.py

```diff
@@ -35,22 +42,29 @@ async def run_evaluator(
     Raises:
         ValueError: If the evaluator returns a value of an invalid type.
     """
-    raw_results = await evaluator.evaluate_async(ctx)
-
     try:
-        results = _EVALUATOR_OUTPUT_ADAPTER.validate_python(raw_results)
-    except ValidationError as e:
-        raise ValueError(f'{evaluator!r}.evaluate returned a value of an invalid type: {raw_results!r}.') from e
-
-    results = _convert_to_mapping(results, scalar_name=evaluator.get_default_evaluation_name())
-
-    details: list[EvaluationResult] = []
-    for name, result in results.items():
-        if not isinstance(result, EvaluationReason):
-            result = EvaluationReason(value=result)
-        details.append(EvaluationResult(name=name, value=result.value, reason=result.reason, source=evaluator))
-
-    return details
+        raw_results = await evaluator.evaluate_async(ctx)
+
+        try:
+            results = _EVALUATOR_OUTPUT_ADAPTER.validate_python(raw_results)
+        except ValidationError as e:
+            raise ValueError(f'{evaluator!r}.evaluate returned a value of an invalid type: {raw_results!r}.') from e
+
+        results = _convert_to_mapping(results, scalar_name=evaluator.get_default_evaluation_name())
+
+        details: list[EvaluationResult] = []
+        for name, result in results.items():
+            if not isinstance(result, EvaluationReason):
+                result = EvaluationReason(value=result)
+            details.append(EvaluationResult(name=name, value=result.value, reason=result.reason, source=evaluator))
+
+        return details
+    except Exception as e:
+        return [
+            EvaluatorFailure(
+                name=evaluator.get_default_evaluation_name(), error_msg=f'{type(e).__name__}: {e}', source=evaluator
```

Comment:
Can we attach the exception itself so it can be read programmatically to get the stack trace if so desired? It wouldn't be serialized of course.